test/clojure/core_test/peek.cljc

```diff
@@ -0,0 +1,32 @@
+(ns clojure.core-test.peek
+  (:require [clojure.test :refer [deftest testing is are]]
+            [clojure.core-test.portability #?(:cljs :refer-macros :default :refer) [when-var-exists]])
+  #?(:clj  (:import [clojure.lang PersistentQueue])
+     :cljr (:import [clojure.lang PersistentQueue])))
+
+(when-var-exists clojure.core/peek
+  (deftest test-peek
+
+    (testing "list and queue"
+      (is (nil? (peek '())))
+      (is (= :a (peek '(:a :b :c))))
+      #?(:cljs    (is (nil? (peek #queue [])))
+         :default (is (nil? (peek PersistentQueue/EMPTY))))
+      #?(:cljs    (is (= :a (peek #queue [:a :b :c])))
+         :default (is (= :a (peek (conj PersistentQueue/EMPTY :a :b :c))))))
```

Comment:
Get what you're going for here (I had a similar suggestion for `pop`) but after talking with @jeaye I think we're better off not testing `clojure.lang.PersistentQueue` as most platforms don't support it and it's not something `clj`/`cljr` "easily" expose (i.e. there's not a way to naturally construct the structure with some `clojure.core` function like `(queue 0 1 2)`)

Reply:
Removed tests on queues.